blocks/table/src/components/table/table.tsx

```diff
@@ -153,41 +170,104 @@ export const Table = ({ blockEntity, updateEntity, readonly }: TableProps) => {
       }, 100);
     });
-  };
+  }, [localColumns, updateStateAndEntity]);
 
-  const addNewRow = () => {
-    const newLocalRows = [...rows, { rowId: `${rows.length}` }];
+  const handleAddNewRow = useCallback(() => {
+    const newLocalRows = [...localRows, {}];
     updateStateAndEntity({ newLocalRows });
-  };
+  }, [localRows, updateStateAndEntity]);
+
+  const handleCellsEdited = useCallback<
+    NonNullable<DataEditorProps["onCellsEdited"]>
+  >(
+    (newValues) => {
+      const mergedLocalRows =
+        numberOfRows > localRows.length
+          ? [
+              ...localRows,
+              ...Array<Object>(numberOfRows - localRows.length).fill(
+                {} as Object,
+              ),
+            ]
+          : localRows;
 
-  const handleCellsEdited: DataEditorProps["onCellsEdited"] = (newValues) => {
-    const newLocalRows = produce(rows, (draftRows) => {
-      for (const { value, location } of newValues) {
-        const [colIndex, rowIndex] = location;
+      const newLocalRows = produce(mergedLocalRows, (draftRows) => {
+        for (const { value, location } of newValues) {
+          const [colIndex, rowIndex] = location;
 
-        const columnId = columns[colIndex]?.id;
+          const columnId = columns[colIndex]?.id;
 
-        if (columnId) {
-          // @ts-expect-error -- type instantiation is deep and possibly infinite
-          draftRows[rowIndex][columnId] = value.data!;
+          if (columnId) {
+            // @ts-expect-error -- type instantiation is deep and possibly infinite
+            draftRows[rowIndex][columnId] = value.data!;
+          }
         }
-      }
-    });
+      });
+
+      updateStateAndEntity({ newLocalRows });
+
+      return true;
+    },
+    [columns, localRows, numberOfRows, updateStateAndEntity],
+  );
+
+  const handleDeleteSelectedRows = useCallback(() => {
+    const selectedRows = selection.rows.toArray();
+
+    const filteredRows = localRows.filter(
+      (_, index) => !selectedRows.includes(index),
+    );
+
+    const newLocalRows = filteredRows.length > 0 ? filteredRows : [{}];
 
     updateStateAndEntity({ newLocalRows });
 
-    return true;
-  };
+    setSelection(emptySelection);
+  }, [localRows, selection, updateStateAndEntity]);
 
-  const getRowThemeOverride: DataEditorProps["getRowThemeOverride"] = (row) => {
-    if (!isStriped) {
-      return undefined;
-    }
+  const getCellContent = useCallback<DataEditorProps["getCellContent"]>(
+    ([colIndex, rowIndex]) => {
+      const key = columns[colIndex]?.id;
 
-    return row % 2 ? { bgCell: "#f9f9f9" } : undefined;
-  };
+      if (!key) {
+        return {
+          kind: GridCellKind.Text,
+          displayData: "",
+          data: "",
+          allowOverlay: false,
+        };
+      }
+
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- todo fix this
+      const value = ((localRows[rowIndex] as any)?.[key] ?? "") as JsonValue;
```

Comment:
One alternative, although `value` ends up inferred as `""` (because no keys exist on `{}`) 🤷 
```suggestion
      const row = localRows[rowIndex];
      const value = row && key in row ? row[key as keyof typeof row] : "";
```